openhtf/util/data.py

```diff
@@ -135,3 +136,31 @@ def ConvertToBaseTypes(obj, ignore_keys=tuple()):
     obj = str(obj)
 
   return obj
+
+
+def TotalSize(obj):
+  """Returns the approximate total memory footprint an object."""
+  seen = set()
+  def sizeof(current_obj):
+    """Do a depth-first acyclic traversal of all reachable objects."""
+    if id(current_obj) in seen:
+      # A rough approximation of the size cost of an additional reference.
+      return struct.calcsize('P')
+    seen.add(id(current_obj))
+    current_size = sys.getsizeof(current_obj)
```

Comment:
current_size is unused, actually, i think i meant to change all instances of `size` to `current_size` and messed up the refactor.